pkg/config/validation.go

```diff
@@ -209,30 +209,41 @@ func validatePodIntegrationOptions(c *configapi.Configuration) field.ErrorList {
 		return allErrs
 	}
 
-	if c.Integrations.PodOptions == nil {
-		return field.ErrorList{field.Required(podOptionsPath, "cannot be empty when pod integration is enabled")}
-	}
-	if c.Integrations.PodOptions.NamespaceSelector == nil {
-		return field.ErrorList{field.Required(namespaceSelectorPath, "a namespace selector is required")}
-	}
-
-	prohibitedNamespaces := []labels.Set{{corev1.LabelMetadataName: metav1.NamespaceSystem}}
-
-	if c.Namespace != nil && *c.Namespace != "" {
-		prohibitedNamespaces = append(prohibitedNamespaces, labels.Set{corev1.LabelMetadataName: *c.Namespace})
-	}
-
-	allErrs = append(allErrs, validation.ValidateLabelSelector(c.Integrations.PodOptions.NamespaceSelector, validation.LabelSelectorValidationOptions{}, namespaceSelectorPath)...)
-
-	selector, err := metav1.LabelSelectorAsSelector(c.Integrations.PodOptions.NamespaceSelector)
-	if err != nil {
+	validateNamespaceSelector := func(namespaceSelector *metav1.LabelSelector, namespaceSelectorPath *field.Path, allErrs field.ErrorList) field.ErrorList {
+		allErrs = append(allErrs, validation.ValidateLabelSelector(namespaceSelector, validation.LabelSelectorValidationOptions{}, namespaceSelectorPath)...)
+		selector, err := metav1.LabelSelectorAsSelector(namespaceSelector)
+		if err != nil {
+			return allErrs
+		}
+		prohibitedNamespaces := []labels.Set{{corev1.LabelMetadataName: metav1.NamespaceSystem}}
+		if c.Namespace != nil && *c.Namespace != "" {
+			prohibitedNamespaces = append(prohibitedNamespaces, labels.Set{corev1.LabelMetadataName: *c.Namespace})
+		}
+		for _, pn := range prohibitedNamespaces {
+			if selector.Matches(pn) {
+				allErrs = append(allErrs, field.Invalid(namespaceSelectorPath, namespaceSelector,
+					fmt.Sprintf("should not match the %q namespace", pn[corev1.LabelMetadataName])))
+			}
+		}
 		return allErrs
 	}
 
-	for _, pn := range prohibitedNamespaces {
-		if selector.Matches(pn) {
-			allErrs = append(allErrs, field.Invalid(namespaceSelectorPath, c.Integrations.PodOptions.NamespaceSelector,
-				fmt.Sprintf("should not match the %q namespace", pn[corev1.LabelMetadataName])))
+	if features.Enabled(features.ManagedJobsNamespaceSelector) {
```

Comment:
This validation is a bit hard to follow, can you summarize what happens if one configures both `c.Integrations.PodOptions` and `c.ManagedJobsNamespaceSelector`. Is this a supported configuration?

Reply:
Configuring both is a supported configuration (although not one we document explicitly).  This is mainly to support upgrading systems that were using the pod integration on prior releases.  The semantics are an and of the two for the Pod-based integrations. If both selectors are present, then if either one fails to match the Pod is exempted from management: 

I'll add a comment and try to structure the validation code a little differently and see if that helps.

Reply:
Reworked the validation logic and added a comment.  I think it's better, so resolving.  Please unresolve if more is needed.

Reply:
lgtm